tests/test_solvers/test_harbor_solver.py

```
@@ -1198,3 +1199,281 @@ async def test_nonexistent_skill_dir_warns_but_continues(self, caplog):
+        docs = [
+            {"steps": [{"source": "agent", "metrics": {"prompt_tokens": 0, "completion_tokens": 0}}]},
+            {"steps": [{"source": "agent", "metrics": {"prompt_tokens": 0, "completion_tokens": 0}}]},
+        ]
+        assert _count_zero_token_agent_turns(docs) == 2
+
+
+# ── OpenHands condenser config plumbing ──────────────────────────────────
+
+
+class TestHarborSolverCondenser:
+    def _make(self, agent="openhands-sdk", agent_kwargs=None, container_env=None):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_make`

(ANN202)